pkgs/stdenv/generic/check-meta.nix

```
@@ -86,12 +89,41 @@ let
     allowInsecurePredicate attrs ||
     builtins.getEnv "NIXPKGS_ALLOW_INSECURE" == "1";
 
-  showLicense = license: toString (map (l: l.shortName or "unknown") (lib.lists.toList license));
+  hasSourceProvenance = attrs:
+    attrs ? meta.sourceProvenance;
+
+  isNonSource = sourceTypes: lib.lists.any (t: !t.isSource or true) sourceTypes;
+
+  hasNonSourceProvenance = attrs:
+    hasSourceProvenance attrs &&
```

```suggestion
    (attrs ? meta.sourceProvenance) &&
```

The name `hasSourceProvenance` is sort of ambiguous: "has `meta.sourceProvenance`" vs "has source-code provenance".

Since `hasSourceProvenance` is used in only one place, why not just inline it?